src/charts/heatmap/heatmap.directive.js

```diff
@@ -0,0 +1,223 @@
+       {'id': 51, 'value': 0.22, 'tooltip': 'Node 26 : My Kubernetes Provider<br\>22% : 22 Used of 100 Total<br\>78 Available'},
+       {'id': 14, 'value': 0.2, 'tooltip': 'Node 14 : My OpenShift Provider<br\>20% : 20 Used of 100 Total<br\>80 Available'}];
+
+       $scope.titleAlt = 'Utilization - Overriding Defaults';
+       $scope.legendLabels = ['< 60%','70%', '70-80%' ,'80-90%', '> 90%'];
+       $scope.thresholds = [0.6, 0.7, 0.8, 0.9];
+       $scope.heatmapColorPattern = ['#d4f0fa', '#F9D67A', '#EC7A08', '#CE0000', '#f00'];
+
+       var clickAction = function (block) {
+          console.log(block);
+       };
+       $scope.clickAction = clickAction;
+     });
+   </file>
+   <file name="index.html">
+     <div ng-controller="ChartCtrl" class="row">
+       <div class="col-md-4">
+         <div pf-heatmap id="id" chart-title="title" data="data"></div>
+       </div>
+       <div class="col-md-4">
+         <div pf-heatmap id="id" chart-title="titleAlt" data="data" legend-labels="legendLabels" heatmap-color-pattern="heatmapColorPattern" thresholds="thresholds" click-action="clickAction"></div>
+       </div>
+     </div>
+   </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfHeatmap', function ($compile) {
+  'use strict';
+  return {
+    restrict: 'A',
+    scope: {
+      data: '=',
+      height: '=',
+      chartTitle: '=?',
+      legendLabels: '=?',
+      thresholds: '=?',
+      heatmapColorPattern: '=?',
+      clickAction: '=?'
+    },
+    replace: true,
+    templateUrl: 'charts/heatmap/heatmap.html',
+    link: function (scope, element, attrs) {
+      var thisComponent = element[0].querySelector('.pf-heatmap-svg');
+      var containerWidth, containerHeight, blockSize, numberOfRows;
+      var thresholdDefaults = [0.7, 0.8, 0.9];
+      var heatmapColorPatternDefaults = ['#d4f0fa', '#F9D67A', '#EC7A08', '#CE0000'];
+      var legendLabelDefaults = ['< 70%', '70-80%' ,'80-90%', '> 90%'];
+      var heightDefault = 200;
+
+      var setSizes = function () {
+        var parentContainer = element[0].querySelector('.heatmap-container');
+        containerWidth = parentContainer.clientWidth;
+        containerHeight = parentContainer.clientHeight;
+        blockSize = determineBlockSize();
+        numberOfRows = (blockSize === 0) ? 0 : Math.floor(containerHeight / blockSize);
+      };
+
+      var determineBlockSize = function () {
+        var x = containerWidth;
+        var y = containerHeight;
+        var n = scope.data.length;
+        var px = Math.ceil(Math.sqrt(n * x / y));
+        var py = Math.ceil(Math.sqrt(n * y / x));
+        var sx, sy;
+
+        if (Math.floor(px * y / x) * px < n) {
+          sx = y / Math.ceil(px * y / x);
+        } else {
+          sx = x / px;
+        }
+
+        if (Math.floor(py * x / y) * py < n) {
+          sy = x / Math.ceil(x * py / y);
+        } else {
+          sy = y / py;
+        }
+        return Math.max(sx, sy);
```

Comment:
I think we should have a maximum size. If there are only 4 blocks, it's not great to draw each of them really huge. This might be an enhancement request.